File: BubbleSort.js
// Array: Bubble Sort
// For review, create a function that uses BubbleSort to sort an unsorted array in -place.”
//Excerpt From: Martin Puryear.“Algorithm Challenges: E - book for Dojo Students.” iBooks.

var arr = [7, 5, 3, 4, 2, 8, 1]

function bubblesort(arrInput) {
    // for j in Range(0, len(arrInput), 1):
    for (var j = 0; j < arrInput.length; j++) {
        for (var i = 0; i < arrInput.length - 1 - j; i++) {
            if (arrInput[i] > arrInput[i + 1]) {
                var temp = arrInput[i]
                arrInput[i] = arrInput[i + 1]
                arrInput[i + 1] = temp
            }
        }
    }
    return arrInput

}


console.log(bubblesort([7, 5, 3, 4, 2, 8, 1]))

File: BubbleSort.py
# Array: Bubble Sort
# For review, create a function that uses BubbleSort to sort an unsorted array in -place.”

# Excerpt From: Martin Puryear. “Algorithm Challenges: E - book for Dojo Students.” iBooks.

# PYTHON
def BubbleSort(arr):
    for i in range(0, len(arr), 1):
        for j in range(0, (len(arr) - i - 1), 1):
            if arr[j] > arr[j + 1]:
                temp = arr[j]
                arr[j] = arr[j+1]
                arr[j+1] = temp

    return arr


arr2 = [2, 7, 1, 44, 25]
print(BubbleSort(arr2))
